Wbooks/src/redux/authentication/actions.js

```diff
@@ -0,0 +1,76 @@
+import { AsyncStorage } from 'react-native';
+
+import AuthenticationService from '../../services/AuthenticationService';
+
+export const actionTypes = {
+  LOG_IN: 'LogIn',
+  LOG_IN_SUCCESS: 'LogInSuccess',
+  LOG_IN_ERROR: 'LogInError',
+  LOADING_ENABLED: 'LoadingEnabled',
+  LOADING_DISABLED: 'LoadingDisabled',
+  STORED_DATA_SUCESS: 'StoredDataSuccess'
+};
+
+const SESSION_DATA_KEY = 'sessionData';
+
+const AuthenticationActions = {
+  startUp: navigation => async dispatch => {
+    try {
+      const sessionData = await AsyncStorage.getItem('sessionData');
+      if (sessionData !== null) {
+        dispatch({
+          type: actionTypes.STORED_DATA_SUCESS,
+          payload: sessionData
+        });
+        navigation.navigate('Main');
+      }
+    } catch (error) {
+      dispatch({
+        type: actionTypes.LOG_IN_ERROR
+      });
+    }
+  },
+
+  logIn: (email, password, navigation) => async dispatch => {
+    dispatch({
+      type: actionTypes.LOADING_ENABLED
+    });
+    const response = await AuthenticationService.signIn(email, password);
+    if (response.ok) {
+      const { data: user } = response.data;
```

Comment:
Here you can keep destructuring data like this `const { data: { uid, email } } = response.data;`. With this you just get what you need from response.data and in just one access, the negative thing here is that if you want to access response data you have to do it like this `response.data.data` now